Assets/C#/EnemyScripts/BaseEnemy.cs

```diff
@@ -0,0 +1,51 @@
+﻿using UnityEngine;
+using System.Collections;
+
+abstract public class BaseEnemy : MonoBehaviour {
+
+	public GameObject [] drops; 	//Array of possible drops for this enemy
+
+	public int minDrops;			//Mininum number of possible drops
+	public int maxDrops;			//Maximum number of possible drops
+	public float health;			//Enemy health
+	public float movementSpeed;		//Enemy movement speed
+
+	/*
+	 * Method called when enemy dies
+	 * Identifies the number of drops the enemy drops based on the minimum and maximum values
+	 * Then, for each of the individual drops, find a random GameObject in the drops array to drop
+	 * Instantiate that drop
+	 */
+	void OnDeath(){
+		int numberOfDrops = Mathf.RoundToInt (Random.Range (minDrops, maxDrops));
+
+		for (int i = 0; i < numberOfDrops; i++) {
+			int dropIndex = Mathf.RoundToInt (Random.Range (0, drops.Length));
+			Instantiate (drops [dropIndex], transform.position, Quaternion.identity);
+		}
+	}
+
+	/*
+	 * The damage done to the enemies health based on the parameter defined damage
+	 */
+	public void TakeDamage (float dmg){
+		health -= dmg;
+	}
+
+	/*
+	 * Each frame, call enemies movement method if it's not attacking and attack method if it is
+	 * Also check if health is below or equal to zero and call OnDeath if true
+	 * note: has to be public because it needs to be in order for the sub-class to use it
+	 */
+	public void Update () {
+		
+		Movement ();
+
+		if (health <= 0) {
+			OnDeath ();
```

Comment:
As it stands, if the enemy takes enough damage, it will constantly call OnDeath(). Also, they will still have movement. Consider moving calling on Death to TakeDamage, and only if the original health > 0 and new health <= 0